discovery-provider/src/queries/get_health.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import os
+import sqlalchemy
+
+from src.models import Block
+from src.utils import helpers, redis_connection, web3_provider, db_session
+from src.utils.config import shared_config
+from src.utils.redis_constants import latest_block_redis_key, \
+    latest_block_hash_redis_key, most_recent_indexed_block_hash_redis_key, most_recent_indexed_block_redis_key
+
+
+logger = logging.getLogger(__name__)
+disc_prov_version = helpers.get_discovery_provider_version()
+
+default_healthy_block_diff = int(
+    shared_config["discprov"]["healthy_block_diff"])
+
+# Returns DB block state & diff
+def _get_db_block_state():
+    db = db_session.get_db_read_replica()
+    with db.scoped_session() as session:
+        # Fetch latest block from DB
+        db_block_query = session.query(Block).filter(
+            Block.is_current == True).all()
+        assert len(db_block_query) == 1, "Expected SINGLE row marked as current"
+        return helpers.model_to_dictionary(db_block_query[0])
+
+
+# Returns number of and info on open db connections
+def _get_db_conn_state():
+    db = db_session.get_db_read_replica()
+    with db.scoped_session() as session:
+        # Query number of open DB connections
+        num_connections = session.execute(
+            sqlalchemy.text("select sum(numbackends) from pg_stat_database;")
+        ).fetchall()
+
+        if not (num_connections and num_connections[0][0]):
+            return 'pg_stat_database query failed', True
+        num_connections = num_connections[0][0]
+
+        # Query connection info
+        connection_info = session.execute(sqlalchemy.text(
+            "select datname, state, query, wait_event_type, wait_event from pg_stat_activity where state is not null;"
+        )).fetchall()
+        connection_info = [dict(row) for row in connection_info]
+
+    return {"open_connections": num_connections, "connection_info": connection_info}, False
+
+
+def get_health(args, use_redis_cache=True):
+    """
+    Gets health status for the service
+
+    :param args: dictionary
+    :param args.verbose: string
+        if 'true', returns db connection information
+    :param args.healthy_block_diff: string (int-like)
+        determines the point at which a block difference is considered unhealthy
+    :param args.enforce_block_diff: string if 'true', and the block difference is unhealthy
```

Comment:
General question about how we want to structure logic for interpreting the params. 
I think it would be cleaner to force each function to be passed args of the correct type so that the function does not have to also check/cast type. 
For the api/v1 I tried to do that so int/bool/list of int types would be parsed & cast before calling the query method.
Also the boolean query param is not a standard, so we should parse that consistency across our api.

Reply:
Yeah -- i thought we were gonna put on this, didn't realize you & michael did a bit more w/ v1 there. Will change in this PR